daml-foundations/daml-tools/daml-extension/src/extension.ts

```diff
@@ -266,8 +266,12 @@ namespace DamlWorkspaceValidationsNotification {
 }
 
 class VirtualResourceManager {
+    // Mapping from URIs to the web view panel
     private _panels: Map<string, vscode.WebviewPanel> = new Map<string, vscode.WebviewPanel>();
+    // Mapping from URIs to the HTML content of the webview
     private _panelContents: Map<string, string> = new Map<string, string>();
```

Comment:
Is there no `Uri` type better than `string`?

Reply:
I actually tried to use `Map<Uri, …>` initially and spend over an hour trying to figure out why it didn’t work until I figured out that this uses reference equality with no way to override it as far as I can tell.

Reply:
I’ll add a comment when I touch this again.

Reply:
I forgot that some languages are completely broken and not fit for purpose...